fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/OnlineIndexer.java

```diff
@@ -2080,7 +2125,12 @@ public Builder toBuilder() {
                     .setIfDisabled(ifDisabled)
                     .setIfWriteOnly(ifWriteOnly)
                     .setIfMismatchPrevious(ifMismatchPrevious)
-                    .setIfReadable(ifReadable);
+                    .setIfReadable(ifReadable)
+                    .allowUniquePendingState(allowUniquePendingState)
+                    .allowTakeoverContinue(allowUniquePendingState)
+                    .setMutualIndexing(mutualIndexing)
+                    .setMutualIndexing(mutualIndexingBoundaries)
```

Comment:
Does this do the wrong thing if someone has set `mutualIndexing` to `true` but left `mutualIndexingBoundaries` as `null`? I believe that's a legal configuration, that would result in the index being built using the primary key shard boundaries, but I also think that the builder copying would set `useMutualIndexing` to `true` in the call to `.setMutualIndexing(mutualIndexing /* = true */)`, and then it would erase that value when calling `.setMutualIndexing(mututalIndexingBoundaries /* = null */)`.

It might be easier to make those methods more conventional setters rather than having `.setMutualIndexing(List)` intuit whether `mutualndexing` should be `true` or `false`.

Reply:
I reversed the order. Would you approve the logic?

Reply:
I see. I can rename the upper function to `setMutualIndexingBoundaries(@Nullable set)`. If non-null, it applies `setMutualIndexing(true)`. Else, keep it as is. 
Will that work?

Reply:
I think that seems more reasonable. I'd be against a change that still subtly relied on the order in which the builder methods are called, but providing a non-`null` set also implicitly enable mutual indexing seems more defensible. I think it does mean that if a user did something like:

```java
builder.setMututalIndexingBoundaries(nonNullSet) // mutual indexing true
    .setMututalIndexing(false) // mutual indexing false
    .build()
    .toBuilder() // mutual indexing may be set to true?
```

Which could be confusing. The alternative, I think, would be to make `.setMutualIndexingBoundaries` _only_ set the boundaries, which does mean that the user needs to set both, which could seem redundant.

Another alternative would be to have `.toBuilder` use a private constructor internally that directly copies the values rather than calling `.setX` for all of the methods so that we're more sure that the internal state is directly being copied

Reply:
Changed to `setMutualIndexingBoundaries`.

Reply:
Also `setMututalIndexing(false)` will clear the boundaries list.